source/common/filter/config_discovery_impl.h

```diff
@@ -66,27 +65,28 @@ class DynamicFilterConfigProviderImplBase
 class DynamicFilterConfigProviderImpl : public DynamicFilterConfigProviderImplBase,
                                         public DynamicFilterConfigProvider {
 public:
-  DynamicFilterConfigProviderImpl(FilterConfigSubscriptionSharedPtr& subscription,
-                                  const absl::flat_hash_set<std::string>& require_type_urls,
-                                  Server::Configuration::FactoryContext& factory_context,
-                                  Envoy::Http::FilterFactoryCb default_config,
-                                  bool last_filter_in_filter_chain,
-                                  const std::string& filter_chain_type)
+  DynamicFilterConfigProviderImpl(
+      FilterConfigSubscriptionSharedPtr& subscription,
+      const absl::flat_hash_set<std::string>& require_type_urls,
+      Server::Configuration::FactoryContext& factory_context,
+      absl::optional<ProtobufWkt::Any> default_config, bool last_filter_in_filter_chain,
```

Comment:
Can `default_config` be passed by reference? and also const?

Reply:
The constructed object needs ownership. Seems I should either change `validateDefaultConfig` back to `get`, returning an `optional` so copy elision can be done, or use `unique_ptr`?

Reply:
To clarify, `getDefaultConfig` would return `absl::optional<ProtobufWkt::Any>` and the function call below would become
```
auto provider = std::make_unique<DynamicFilterConfigProviderImpl>(..., getDefaultConfig(), ...);
```
to avoid the copy. But I'm not sure if copy elision is discouraged in Envoy style vs. just allocating on the heap

Reply:
I might be missing something here, but isn't `default_config` copied into `default_configuration_`?

Reply:
Sorry, for some reason I thought you were trying to avoid that copy too. I updated to use a const reference